extensions-contrib/momentsketch/src/main/java/org/apache/druid/query/aggregation/momentsketch/aggregator/MomentSketchAggregatorFactory.java

```diff
@@ -0,0 +1,295 @@
+
+  @Override
+  public Aggregator factorize(ColumnSelectorFactory metricFactory)
+  {
+    ColumnCapabilities cap = metricFactory.getColumnCapabilities(fieldName);
+    if (cap == null || ValueType.isNumeric(cap.getType())) {
+      final ColumnValueSelector<Double> selector = metricFactory.makeColumnValueSelector(fieldName);
+      return new MomentSketchBuildAggregator(selector, k, getCompress());
+    } else {
+      final ColumnValueSelector<MomentSketchWrapper> selector = metricFactory.makeColumnValueSelector(fieldName);
+      return new MomentSketchMergeAggregator(selector, k, getCompress());
+    }
+  }
+
+  @Override
+  public BufferAggregator factorizeBuffered(ColumnSelectorFactory metricFactory)
+  {
+    ColumnCapabilities cap = metricFactory.getColumnCapabilities(fieldName);
+    if (cap == null || ValueType.isNumeric(cap.getType())) {
+      final ColumnValueSelector<Double> selector = metricFactory.makeColumnValueSelector(fieldName);
+      return new MomentSketchBuildBufferAggregator(selector, k, getCompress());
+    } else {
+      final ColumnValueSelector<MomentSketchWrapper> selector = metricFactory.makeColumnValueSelector(fieldName);
+      return new MomentSketchMergeBufferAggregator(selector, k, getCompress());
+    }
+  }
+
+  public static final Comparator<MomentSketchWrapper> COMPARATOR = Comparator.nullsFirst(
+      Comparator.comparingDouble(a -> a.getPowerSums()[0])
+  );
+
+  @Override
+  public Comparator getComparator()
+  {
+    return COMPARATOR;
+  }
+
+  @Override
+  public Object combine(@Nullable Object lhs, @Nullable Object rhs)
+  {
+    if (lhs == null) {
+      return rhs;
+    }
+    if (rhs == null) {
+      return lhs;
+    }
+    MomentSketchWrapper union = (MomentSketchWrapper) lhs;
+    union.merge((MomentSketchWrapper) rhs);
+    return union;
+  }
+
+  @Override
+  public AggregatorFactory getCombiningFactory()
+  {
+    return new MomentSketchMergeAggregatorFactory(name, k, compress);
+  }
+
+  @Override
+  public AggregatorFactory getMergingFactory(AggregatorFactory other) throws AggregatorFactoryNotMergeableException
+  {
+    if (other.getName().equals(this.getName()) && this.getClass() == other.getClass()) {
+      return getCombiningFactory();
+    } else {
+      throw new AggregatorFactoryNotMergeableException(this, other);
+    }
+  }
+
+  @Override
+  public List<AggregatorFactory> getRequiredColumns()
+  {
+    return Collections.singletonList(
+        new MomentSketchAggregatorFactory(
+            fieldName,
+            fieldName,
+            k,
+            compress
+        )
+    );
+  }
+
+  private MomentSketchWrapper deserializeFromByteArray(byte[] bytes)
+  {
+    return MomentSketchWrapper.fromByteArray(bytes);
+  }
+
+  @Override
+  public Object deserialize(Object serializedSketch)
+  {
+    if (serializedSketch instanceof String) {
+      String str = (String) serializedSketch;
+      return deserializeFromByteArray(Base64.decodeBase64(str.getBytes(Charsets.UTF_8)));
```

Comment:
This should use `StringUtils.decodeBase64` and `StringUtils.toUtf8` to pass the style checks